packages/vite/src/node/server/environment.ts

```diff
@@ -118,16 +116,15 @@ export class DevEnvironment extends BaseEnvironment {
       this.pluginContainer!.resolveId(url, undefined),
     )
 
-    this.hot = setup.hot || createNoopHotChannel()
-    this.watcher = setup.watcher
+    this.hot = context.hot || createNoopHotChannel()
 
     this._onCrawlEndCallbacks = []
     this._crawlEndFinder = setupOnCrawlEnd(() => {
       this._onCrawlEndCallbacks.forEach((cb) => cb())
     })
 
-    this._ssrRunnerOptions = setup.runner || {}
-    setup.runner?.transport?.register(this)
+    this._ssrRunnerOptions = context?.runner ?? {}
```

Comment:
?. not needed anymore

Reply:
Ah, nice catch, the diff is strange here. I'll commit the change.